From 5febdf595e514be41480a997cc606570e704fb6c Mon Sep 17 00:00:00 2001
From: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
Date: Mon, 28 Sep 2020 12:11:27 +0200
Subject: [PATCH] Fix compilation after mime data API changes

Follows 848927b688d4709d0b71590518e2c32fe7866147 in qtbase,
where the signature of retrieveData_sys() was changed.

Fixes: QTBUG-87003
Change-Id: I8b88c8f9547e599328f40345a9fbf57c07ec31a4
---
 src/client/qwaylanddataoffer.cpp | 2 +-
 src/client/qwaylanddataoffer_p.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/client/qwaylanddataoffer.cpp b/src/client/qwaylanddataoffer.cpp
index 2297e8a1..5d0a8d47 100644
--- a/src/client/qwaylanddataoffer.cpp
+++ b/src/client/qwaylanddataoffer.cpp
@@ -124,7 +124,7 @@ QStringList QWaylandMimeData::formats_sys() const
     return m_types;
 }
 
-QVariant QWaylandMimeData::retrieveData_sys(const QString &mimeType, QVariant::Type type) const
+QVariant QWaylandMimeData::retrieveData_sys(const QString &mimeType, QMetaType type) const
 {
     Q_UNUSED(type);
 
diff --git a/src/client/qwaylanddataoffer_p.h b/src/client/qwaylanddataoffer_p.h
index 9cf1483c..50c05732 100644
--- a/src/client/qwaylanddataoffer_p.h
+++ b/src/client/qwaylanddataoffer_p.h
@@ -106,7 +106,7 @@ public:
 protected:
     bool hasFormat_sys(const QString &mimeType) const override;
     QStringList formats_sys() const override;
-    QVariant retrieveData_sys(const QString &mimeType, QVariant::Type type) const override;
+    QVariant retrieveData_sys(const QString &mimeType, QMetaType type) const override;
 
 private:
     int readData(int fd, QByteArray &data) const;
